src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -419,15 +462,21 @@ internal void InvokeResponse(InvocationResponse invokeResponse)
             {
                 try
                 {
-                    IDictionary<string, object> bindingsDictionary = invokeResponse.OutputData
-                        .ToDictionary(binding => binding.Name, binding => binding.Data.ToObject());
+                    IDictionary<string, object> bindingsDictionary = await invokeResponse.OutputData
+                        .ToDictionaryAsync(binding => binding.Name, binding => GetBindingDataAsync(binding));
 
                     var result = new ScriptInvocationResult()
                     {
                         Outputs = bindingsDictionary,
                         Return = invokeResponse?.ReturnValue?.ToObject()
                     };
                     context.ResultSource.SetResult(result);
+
+                    // List of shared memory maps that were produced by the worker (for output bindings)
+                    IList<string> outputMaps = GetOutputMaps(invokeResponse.OutputData);
+
+                    // If this invocation was using any shared memory maps, close them to free memory
+                    CloseSharedMemoryResourcesForInvocation(invokeResponse.InvocationId, outputMaps);
```

Comment:
I'm putting that check inside `SendCloseSharedMemoryResourcesRequestRequest`. 
We need to also check if any *input* maps were allocated for this invocation and instead of first checking and then deleting, I'm doing that through `TryFreeSharedMemoryMapsForInvocation` to reduce an extra call.

Reply:
So basically, we call this method `SendCloseSharedMemoryResourcesRequestRequest` for each invocation but it will try to delete any input maps, and also send the message to the worker only `if (outputMaps.Any())` is `true`.